File: backend/CvForge.Api/Domain/LetterDocument.cs
namespace CvForge.Api.Domain;

public class LetterDocument
{
    public string TemplateKey { get; set; } = "classique";
    public LetterSettings Settings { get; set; } = new();
    public PersonalInfo Sender { get; set; } = new();
    public RecipientInfo Recipient { get; set; } = new();
    public string JobTitle { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Date { get; set; } = string.Empty;
    public string Subject { get; set; } = string.Empty;
    public string Introduction { get; set; } = string.Empty;
    public string Motivation { get; set; } = string.Empty;
    public string Skills { get; set; } = string.Empty;
    public string Conclusion { get; set; } = string.Empty;
}

public class RecipientInfo
{
    public string RecruiterName { get; set; } = string.Empty;
    public string CompanyName { get; set; } = string.Empty;
    public string CompanyAddress { get; set; } = string.Empty;
}

public class LetterSettings
{
    public string PrimaryColor { get; set; } = "#0F172A";
    public string SecondaryColor { get; set; } = "#64748B";
    public string FontFamily { get; set; } = "inter";
    public double FontScale { get; set; } = 1.0;
    public string Language { get; set; } = "fr";
}

File: frontend/src/app/models/letter-document.ts
import { PersonalInfo } from './cv-document';

export interface LetterSettings {
  primaryColor: string;
  secondaryColor: string;
  fontFamily: string;
  fontScale: number;
  language: 'fr' | 'en';
}

export interface RecipientInfo {
  recruiterName: string;
  companyName: string;
  companyAddress: string;
}

export interface LetterDocument {
  templateKey: string;
  settings: LetterSettings;
  sender: PersonalInfo;
  recipient: RecipientInfo;
  jobTitle: string;
  city: string;
  date: string;
  subject: string;
  introduction: string;
  motivation: string;
  skills: string;
  conclusion: string;
}

export interface LetterDto {
  id: string;
  name: string;
  document: LetterDocument;
  createdAt: string;
  updatedAt: string;
}

export interface LetterSummary {
  id: string;
  name: string;
  createdAt: string;
  updatedAt: string;
}
